### ISSUE TITLE
ExecAllocator swallows process errors

### ISSUE THREAD

**Issue by tmm1** [chromedp/chromedp#408]
https://github.com/chromedp/chromedp/blob/e5acbe3f56d48fe699363d2ae2f6cefeda9269be/allocate.go#L191-L194

It would be useful to surface errors from `cmd.Wait()` for cases where the process is getting killed due to OOM, segfaulting, etc (essentially, seeing the signal number that killed chrome and/or its exit code is useful).

**Comment by mvdan** [comment]
Yeah, this is planned - I think @kenshaw has a refactor in progress that tackles better exec errors as part of it.

**Comment by kenshaw** [comment]
Yes, this will be done for the next release.

**Comment by tmm1** [comment]
Any updates on this refactor?

**Comment by ZzEeKkAa** [comment]
On long run with multiple chromedp starts this issue leads to goroutine stuck. Possibly, because of unhandled error. Guess it's better to rewrite this part of code in cmd.Run, rather than cmd.Start Here is part of the pprof on goroutines that suck:

```
197 @ 0x42fb4f 0x42b25a 0x42a846 0x4cac0b 0x4cba1b 0x4cb9fc 0x4d2b50 0x4d2b23 0x9753b9 0x974b5f 0x978d7b 0xb71929 0xb6aca2 0x45cf41
#	0x42a845	internal/poll.runtime_pollWait+0x55								/usr/local/Cellar/go/1.12.1/libexec/src/runtime/netpoll.go:182
#	0x4cac0a	internal/poll.(*pollDesc).wait+0x9a								/usr/local/Cellar/go/1.12.1/libexec/src/internal/poll/fd_poll_runtime.go:87
#	0x4cba1a	internal/poll.(*pollDesc).waitRead+0x19a							/usr/local/Cellar/go/1.12.1/libexec/src/internal/poll/fd_poll_runtime.go:92
#	0x4cb9fb	internal/poll.(*FD).Read+0x17b									/usr/local/Cellar/go/1.12.1/libexec/src/internal/poll/fd_unix.go:169
#	0x4d2b4f	os.(*File).read+0x6f										/usr/local/Cellar/go/1.12.1/libexec/src/os/file_unix.go:263
#	0x4d2b22	os.(*File).Read+0x42										/usr/local/Cellar/go/1.12.1/libexec/src/os/file.go:108
#	0x9753b8	github.com/chromedp/chromedp.readOutput+0x128							/Users/zeka/go/src/github.com/chromedp/chromedp/allocate.go:239
#	0x974b5e	github.com/chromedp/chromedp.(*ExecAllocator).Allocate+0xa8e					/Users/zeka/go/src/github.com/chromedp/chromedp/allocate.go:205
#	0x978d7a	vendor/github.com/chromedp/chromedp.Run+0x20a							github.com/chromedp/chromedp/chromedp.go:210
```

I will take an eye on it. Once I find stable fix - I'll post PR here

**Comment by ZzEeKkAa** [comment]
Ok, I found root of my issue. It is a little bit different, but related to the same part of code.
https://github.com/chromedp/chromedp/blob/e5acbe3f56d48fe699363d2ae2f6cefeda9269be/allocate.go#L234
See #460 for more details.

BTW: @kenshaw @mvdan why do we need this? 
https://github.com/chromedp/chromedp/blob/e5acbe3f56d48fe699363d2ae2f6cefeda9269be/allocate.go#L189
May be we should send `cancel()` after `cmd.Wait()`. It means that chrome instance is down.

**Comment by mvdan** [comment]
We're not actively working on the API refactor at the moment, so please do send PRs for bugfixes and important features. I already replied to #460 a few days ago.

The `<-ctx.Done()` in that line is because that's a cleanup goroutine; it should only run when the context has been cancelled. Note that Chrome crashing or being closed/killed by the user should *also* cancel the context.

>  May be we should send `cancel()` after `cmd.Wait()`. It means that chrome instance is down.

Not sure why that would be useful; the code will only execute if the context is already cancelled.

**Comment by mvdan** [comment]
Also, if anyone wants to send a PR for this issue in particular, please do; it should be possible to fix it without making any API changes, I think.
